theano/sandbox/cuda/basic_ops.py

```diff
@@ -3150,9 +3138,24 @@ def c_code(self, node, name, inputs, out_, sub):
 
         """ % locals()
 
+        # Test negative axis
+        str += """
+        if( axis < -nd ){
+            char error_message[50];
+            sprintf(error_message, "Join axis %%d out of bounds [0, %%d)",
+                    axis, nd);
+            PyErr_SetString(PyExc_IndexError, error_message);
```

Comment:
Can you use "PyErr_Format(PyExc_IndexError, "Join axis %%d out of bounds [0, %%d)", axis, nd);"

This is the normal way and is simpler.